Packs/CommonScripts/Scripts/SetGridField/SetGridField.py

```diff
@@ -72,33 +115,61 @@ def get_current_table(grid_id: str) -> List[Dict[Any, Any]]:
     return pd.DataFrame(current_table)
 
 
-def validate_entry_context(entry_context: Any, keys: List[str]):
+def validate_entry_context(entry_context: Any, keys: List[str], skip_nested_elements: bool):
     """ Validate entry context structure is valid should be:
             1. List[Dict[str, str]
-            2. Dict[str, str] - for developer it will be in first index of a list.
+            2. List[str/bool/int/float]
+            3. Dict[str, str] - for developer it will be in first index of a list.
 
     Args:
         entry_context: Entry context to validate
         keys: keys to collect data from
+        skip_nested_elements: False for unpacking nested elements, True otherwise.
 
     Raises:
         ValueError: If structure is not valid.
+        data_type (str): The type of information in the context path.
     """
-    exception_msg = "Not valid entry context path - dict[str,str] or list[dict[str,str]]"
-    if not isinstance(entry_context, list):
+    exception_msg = "Not valid entry context path - dict[Any,Any]"
+    if not skip_nested_elements:
+        if not isinstance(entry_context, dict):
+            raise ValueError(exception_msg)
+        else:
+            return
+
+    exception_msg = "Not valid entry context path - dict[str,str] or list[dict[str,str]] or List[str/bool/int/float]"
+    if not isinstance(entry_context, (list, dict)):
         raise ValueError(exception_msg)
 
+    data_type = 'dicts'
+
+    if isinstance(entry_context, dict):
+        return data_type
+
+    has_seen_dict = False
     for item in entry_context:
         if not isinstance(item, dict):
-            raise ValueError(exception_msg)
+            if not has_seen_dict:
+                break
+            else:
+                raise ValueError(exception_msg)
```

Comment:
why is that problematic, we can have dict in the table

Reply:
In the mode of `unpack_nested_elements` set to False, the context object should be either
- dict with primitive values only.
- list with primitive values only.
- list of dicts with primitive values only.

Therefore, this line catches the situation where I have a mix of primitive and non-primitive values in the list.